modules/third-party/org-apache-axis/src/main/java/org/apache/axis/client/ServiceFactory.java

```diff
@@ -106,6 +106,17 @@ public static Service getService(Map environment)
         
         if (context != null) {
             String name = (String)environment.get("jndiName");
+
+            if(name!=null && (name.toUpperCase().indexOf("LDAP")!=-1 ||
+                    name.toUpperCase().indexOf("RMI")!=-1 ||
+                    name.toUpperCase().indexOf("JMS")!=-1 ||
+                    name.toUpperCase().indexOf("JMX")!=-1) ||
+                    name.toUpperCase().indexOf("JRMP")!=-1 ||
+                    name.toUpperCase().indexOf("JAVA")!=-1 ||
+                    name.toUpperCase().indexOf("DNS")!=-1)  {
+
+                return null;
+            }
```

Comment:
@marianoalvarosaiz, the fix referred to https://github.com/apache/axis-axis1-java/commit/7e66753427466590d6def0125e448d2791723210. But the official fix readability is not good, And it existed NPE issue when name = null. So we need to change it and improve readability.